[Bugfix] DeepFace.represent model parameter bugfix and completed parameters docstring

ronaldpereira opened the pull request:
# Description

This PR bugfixes the model parameter to `DeepFace.represent()` and added the complete parameter docstring.

## Current behavior

`DeepFace.represent()` ignores the `model` parameter as it is always overwritten in the code invariantly, making the `model` parameter useless.

```py
model = build_model(model_name)
```

## New behavior

`DeepFace.represent()` only uses the `model_name` parameter and build the `model` only if the `model` parameter wasn't sent (`model = None`).

```py
if model is None:
    model = build_model(model_name)
```

## Tests
- [x] Run tests/unit_test.py

serengil commented:
thank you for your contribution. actually i retired the model input argument in represent function and other functions. but it seems i forgot to drop it in represent function.

ronaldpereira commented:
> thank you for your contribution. actually i retired the model input argument in represent function and other functions. but it seems i forgot to drop it in represent function.

That's ok. I've reverted the change and removed the `model` parameter, but the docstring creating for the other parameters is still necessary, correct?

serengil commented:
yes, other arguments are still valid. thank you again.
